### PR TITLE
Changing OR object to a list of the proper OR class

### PR DESCRIPTION
**Pull request by amnguye**
In ObjectReplication, replacing IDictionary<string, IDictionary<string, string> to IList<ObjectReplicationPolicy>, as per API View comment.
```
IList<ObjectReplicationPolicy> policy;
class ObjectReplicationPolicy
{
   string PolicyId;
   IList<ObjectReplicationRule> Rules;
}
class ObjectReplicationRule
{
  string ruleId;
  ObjectReplicationStatus status;
}
enum ObjectReplicationStatus
{
  Complete,
  Failed
}
```

This will allow the user to know what property of the source OR properties they are accessing as well as giving the room to add more properties in the future.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/Models/ObjectReplicationPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    /// Contains Object Replication Policy ID and the respective list of
+    /// <see cref="ObjectReplicationRule"/>(s). This is used when retrieving the
+    /// Object Replication Properties on the source blob. The policy id for the
+    /// destination blob is set in ObjectReplicationDestinationPolicy of the respective
+    /// method responses. (e.g. <see cref="BlobProperties.ObjectReplicationDestinationPolicy"/>,
+    /// <see cref="BlobDownloadDetails.ObjectReplicationDestinationPolicy"/>).
+    /// </summary>
+    public class ObjectReplicationPolicy
+    {
+        internal ObjectReplicationPolicy() { }
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Models/ObjectReplicationPolicy.cs:19]: I think we have a SomethingFactory class where customers can create these for test purposes. We should add ORS entities there as well to factory method for the type enclosing them.

**Comment by kasobol-msft** [comment]
Looks good (other than Factory thing).

File: sdk/storage/Azure.Storage.Blobs/src/Models/ObjectReplicationPolicy.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Models/ObjectReplicationPolicy.cs:19]: @amnguye you can add methods to create `ObjectReplicationPolicy` and `ObjectReplicationRule` in the  `BlobsModelFactory`.

**Review by seanmcc-msft — APPROVED** [review]
Let's add the `BlobModelFactory` methods.

File: sdk/storage/Azure.Storage.Blobs/src/Models/BlobsModelFactory.cs [hunk 2]
```diff
@@ -163,7 +163,7 @@ public partial class BlobsModelFactory
             DateTimeOffset accessTierChangedOn,
             BlobType blobType,
             string versionId,
-            IDictionary<string, IDictionary<string, string>> objectReplicationSourceProperties,
+            IList<ObjectReplicationPolicy> objectReplicationSourceProperties,
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Models/BlobsModelFactory.cs:166]: We should also add this to BlobDownloadDetails's factory.

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**